internal/cli/run.go

```diff
@@ -111,68 +123,113 @@ func App(opts *common.CLIOpts) *cli.App {
 			Value:   false,
 			Usage:   "EXPERIMENTAL: watch config files for changes and automatically apply them",
 		},
+		&cli.StringSliceFlag{
+			Name:    "env-file",
+			Aliases: []string{"e"},
+			Value:   cli.NewStringSlice(),
+			Usage:   "import environment variables from a dotenv file",
+		},
+		&cli.StringSliceFlag{
+			Name:    "templates",
+			Aliases: []string{"t"},
+			Usage:   opts.ExecTemplate("EXPERIMENTAL: import {{.ProductName}} templates, supports glob patterns (requires quotes)"),
+		},
+	}
+}
+
+func preRun(c *cli.Context, opts *common.CLIOpts) error {
+	dotEnvPaths, err := filepath.Globs(ifs.OS(), c.StringSlice("env-file"))
+	if err != nil {
+		fmt.Printf("Failed to resolve env file glob pattern: %v\n", err)
+		os.Exit(1)
+	}
+	for _, dotEnvFile := range dotEnvPaths {
+		dotEnvBytes, err := ifs.ReadFile(ifs.OS(), dotEnvFile)
+		if err != nil {
+			fmt.Printf("Failed to read dotenv file: %v\n", err)
+			os.Exit(1)
+		}
+		vars, err := parser.ParseDotEnvFile(dotEnvBytes)
+		if err != nil {
+			fmt.Printf("Failed to parse dotenv file: %v\n", err)
+			os.Exit(1)
+		}
+		for k, v := range vars {
+			if err = os.Setenv(k, v); err != nil {
+				fmt.Printf("Failed to set env var '%v': %v\n", k, err)
+				os.Exit(1)
+			}
+		}
+	}
+
+	templatesPaths, err := filepath.Globs(ifs.OS(), c.StringSlice("templates"))
+	if err != nil {
+		fmt.Printf("Failed to resolve template glob pattern: %v\n", err)
+		os.Exit(1)
+	}
+	lints, err := template.InitTemplates(templatesPaths...)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Template file read error: %v\n", err)
+		os.Exit(1)
+	}
+	if !c.Bool("chilled") && len(lints) > 0 {
+		for _, lint := range lints {
+			fmt.Fprintln(os.Stderr, lint)
+		}
+		fmt.Println(opts.ExecTemplate("Shutting down due to linter errors, to prevent shutdown run {{.ProductName}} with --chilled"))
+		os.Exit(1)
+	}
+	return nil
+}
+
+// App returns the full CLI app definition, this is useful for writing unit
+// tests around the CLI.
+func App(opts *common.CLIOpts) *cli.App {
+	flags := []cli.Flag{
+		&cli.BoolFlag{
+			Name:    "version",
+			Aliases: []string{"v"},
+			Value:   false,
+			Usage:   "display version info, then exit",
+		},
+		&cli.StringFlag{
+			Name:    "config",
+			Aliases: []string{"c"},
+			Value:   "",
+			Usage:   "a path to a configuration file",
+		},
+		&cli.BoolFlag{
+			Name:   "help-autocomplete",
+			Value:  false,
+			Usage:  "print json serialised cli argument definitions to assist with autocomplete",
+			Hidden: true,
+		},
 	}
+	flags = append(flags, runFlags(opts)...)
 
 	app := &cli.App{
-		Name:  "Bento",
-		Usage: "A stream processor for mundane tasks - https://warpstreamlabs.github.io/bento",
-		Description: `
-Either run Bento as a stream processor or choose a command:
+		Name:  opts.BinaryName,
+		Usage: opts.ExecTemplate("A stream processor for mundane tasks - {{.DocumentationURL}}"),
```

Comment:
```suggestion
		Usage: opts.ExecTemplate("A stream processor for mundane tasks - {{.DocumentationURL}}/about"),
```

That link is broken - need to append /about to the end.

Reply:
Good catch. Perhaps it would be worthwhile to redirect `/docs` to `/docs/about`